src/xbot/common/logic/Latch.java

```diff
@@ -20,31 +20,13 @@ public Latch(boolean initialValue, EdgeType latchType) {
     }
 
     public void setValue(boolean newValue) {
-        if (this.value == newValue) {
-            return; /* Do nothing if value stays the same */
-        }
+        if (this.value == newValue) return; /* Do nothing if value stays the same */
 
-        this.value = newValue;
         EdgeType edgeType = newValue ? EdgeType.RisingEdge : EdgeType.FallingEdge;
-        boolean alertWatchers;
-        switch (latchType) {
-            case RisingEdge:
-                alertWatchers = newValue; /* Alert Watchers if newValue equals true */
-                break;
-            case FallingEdge:
-                alertWatchers = !newValue; /* Alert Watchers if newValue equals false */
-                break;
-            case Both:
-                alertWatchers = true;
-                break;
-            default:
-                alertWatchers = false;
-                break;
-        }
-
-        if (alertWatchers) {
+        if (latchType == edgeType || latchType == EdgeType.Both) {
```

Comment:
Can you do that? I'm not sure if enum equality works like that in Java...

Reply:
Never mind, yeah that looks right: https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals